enclave/src/error.rs

```diff
@@ -0,0 +1,29 @@
+use derive_more::{Display, From};
+use sgx_types::sgx_status_t;
+use crate::rpc;
+
+#[derive(Debug, Display, From)]
+pub enum Error {
+	Rpc(rpc::error::Error),
+	Codec(codec::Error),
+	Sgx(sgx_status_t)
+}
+
+impl From<Error> for sgx_status_t {
```

Comment:
Implementing `From` does give you `Into` for free, which is what we actually need.